cpp/src/lists/lists_column_factories.cu

```diff
@@ -16,10 +16,62 @@
 
 #include <cudf/column/column.hpp>
 #include <cudf/column/column_factories.hpp>
+#include <cudf/detail/gather.cuh>
 
 #include <rmm/cuda_stream_view.hpp>
+#include <rmm/exec_policy.hpp>
+
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/sequence.h>
 
 namespace cudf {
+namespace lists {
+namespace detail {
+
+std::unique_ptr<cudf::column> make_lists_column_from_scalar(list_scalar const& value,
+                                                            size_type size,
+                                                            rmm::cuda_stream_view stream,
+                                                            rmm::mr::device_memory_resource* mr)
+{
+  if (size == 0) {
+    return make_lists_column(0,
+                             make_empty_column(data_type{type_id::INT32}),
+                             empty_like(value.view()),
+                             0,
+                             cudf::create_null_mask(0, mask_state::UNALLOCATED));
```

Comment:
Couldn't you extend `make_empty_column` so that it works for `LIST` data type? So this could be something like:

```suggestion
    return make_empty_column(data_type{type_id::LIST});
```

Reply:
I think an empty column specifying only `LIST` type is ambiguous - it could be a `List<int>` or `List<List<int>>`. In cudf, the way to distinguish between the two is to store a complete (0-length) column hierarchy. This is kind of similar to the issue I encountered here: https://github.com/rapidsai/cudf/pull/8206

On another note, I couldn't find this being mentioned in the developer guide. Was thinking to submit an issue to discuss.

Reply:
Does an empty lists column ever need to have a hierarchy deeper than a single level? If not, you could add `make_empty_lists_column(data_type contained_data_type)`.

Reply:
I think lists column can be arbitrarily deep. Using `lists_column_wrapper` we can construct an empty list that's of type `List<List<int>>`
```
LCW{LCW{LCW{1, 2, 3}}, LCW{}} // row 1
```

Worse still, there can be `List<Struct<some random fields>>>`, which also cannot be specified using just `type_id::STRUCT`.

Reply:
I think column hierarchy is there to carry the type information of the column. An empty column can still participate in column binary ops. For example, concatenating a `List<List<int>>` column with an empty `List<List<int>>` column is fine, but concatenating with an empty `List<List<List<int>>>` column is an type error. A somewhat contrived example, but the idea extends.